packages/idempotency/src/IdempotencyHandler.ts

```diff
@@ -313,11 +314,22 @@ export class IdempotencyHandler<Func extends AnyFunction> {
         );
       } catch (e) {
         if (e instanceof IdempotencyItemAlreadyExistsError) {
-          const idempotencyRecord: IdempotencyRecord =
-            e.existingRecord ||
-            (await this.#persistenceStore.getRecord(
+          let idempotencyRecord = e.existingRecord;
+          if (idempotencyRecord !== undefined) {
```

Comment:
Just a nitpick: it'd be easier to read if we simplify the logic to `if(idempotencyRecord)`

Reply:
If the `idempotencyRecord` is for whatever reason `false`, or `0`, or `null`, then that statement would also evaluate to `true`  (the beauties of JS) and crash, so I opted to be explicit. 

In other instances where we control the value end to end it's fine to just assert a false-y value, but in this case since we are getting data from a persistence store, I wanted to be on the safe side.